### PR TITLE
Client: Initial client infrastructure for the cryptid anchor rewrite.

### PR DESCRIPTION
**Pull request by dankelleher**
The only changes here are:

A yarn workspaces structure (mostly empty for now)

The IDL types are exported into a package, committed into git, and imported into the tests, so the tests no longer look at the target folder.

### CODE REVIEW

File: cryptid_anchor/Anchor.toml [hunk 1]
```diff
@@ -37,4 +38,4 @@ filename = "../fixtures/did-sol-idl-account.json"
 
 
 [scripts]
-test = "yarn run ts-mocha -p ./tsconfig.json -t 1000000 tests/**/*.ts"
+test = "yarn run ts-mocha --exit -p ./tsconfig.json -t 1000000 tests/**/*.ts"
```
Comment (rado0x54) [inline cryptid_anchor/Anchor.toml:41]:
```
--exit                 Force Mocha to quit after tests complete  [boolean]
```
Why is this needed?

File: cryptid_anchor/client/packages/idl/check_pass.ts [hunk 2]
```diff
@@ -0,0 +1,295 @@
+export type CheckPass = {
```
Comment (rado0x54) [inline cryptid_anchor/client/packages/idl/check_pass.ts:1]: Is this location the default that the anchor command generates to? I'm tempted to not commit this file, but always generate it with a pre-build command.

**Comment by rado0x54** [comment]
Still thinking about the IDL specific packages? Did you see that from other projects? Probably the right way to go, but can they be generated on build and not commited?

File: cryptid_anchor/Anchor.toml (hunk 1 above)
Comment (dankelleher) [reply cryptid_anchor/Anchor.toml:41]: without it, `anchor test` hangs for me after running tests. Not for you?
